### PR TITLE
thorvg: Re-enable thread support, made opt-in upstream

### LINKED ISSUE #87788: Importing SVG's in 4.3.dev with commit 73589f6 is crashing
### Tested versions

- Reverting https://github.com/godotengine/godot/pull/87612 resolves the crashing
- Crash is not always present on windows but sometimes is
- Linux could repro the crash each time booting the engine provided you nuke the .godot/ folder

### System information

Linux - Fedora 38 - As up to date as possible

### Issue description

When importing SVG files in multiple threads the engine crashes with a realloc() crash, I did try to workaround this but had the same crash
image

### Steps to reproduce

- Import a pool of SVG's (doesn't seem to matter the type)

### Minimal reproduction project (MRP)

[TESTPROJECTREPRO.zip](https://github.com/godotengine/godot/files/14113916/TESTPROJECTREPRO.zip)

Images taken from another source, our project files are completely unrelated and they also crash.

### LINKED ISSUE #87962: ThorVG regression causing potentially corrupted SVGs
### Tested versions

- Master (b4e2a24c1f62088b3f7ce0197afc90832fc25009)

### System information

Arch Linux

### Issue description

When rendering SVGs using `ImageLoaderSVG::create_image_from_utf8_buffer` (I'm doing it from C++, but I can't see why it wouldn't affect godot proper too) the rendered images sometimes come out corrupted, an example:

cock_5

I believe it may be a ThorVG bug, since if I downgrade to thorvg 0.12.1 it works fine, but the issue is introduced in 0.12.2

### Steps to reproduce

- Delete .godot folder
- open the project
- svgs may be corrupt, or not, since it's probably some kind of data race it might be unreliable, but at least one of the svgs comes out corrupted for me

### Minimal reproduction project (MRP)

[mrp.zip](https://github.com/godotengine/godot/files/14159009/mrp.zip)

### PR DESCRIPTION
**Pull request by akien-mga**
- Fixup to #87612.
- Fixes #87788.
- Fixes #87962.

### CODE REVIEW

File: thirdparty/thorvg/src/common/tvgLock.h [hunk 1]
```diff
@@ -38,10 +38,10 @@ namespace tvg {
     {
         Key* key = nullptr;
 
-        ScopedLock(Key& key)
+        ScopedLock(Key& p_key)
```
Comment (akien-mga) [inline thirdparty/thorvg/src/common/tvgLock.h:41]: @hermet FYI, I was getting a `-Wshadow` warning from GCC here.
Happy to make a PR upstream if this kind of change looks acceptable, possibly with a different naming convention if there's one you favor.

**Comment by akien-mga** [comment]
I'll yolo merge as I'm about to build 4.3-dev3 and would like this regression fixed in that build.

**Comment by capnm** [comment]
I forgot that this get only default within the meson, sorry.

**Comment by akien-mga** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**